### ISSUE TITLE
Error on installing ruby's dependecies stage in GH Actions on 6.4.2 version

### ISSUE THREAD

**Issue by nukdokplex** [cotes2020/jekyll-theme-chirpy#1499]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Describe the bug

Something went wrong on stage **"Setup Ruby"** while installing dependencies on my website: [nukdokplex/nukdokplex-site](https://github.com/nukdokplex/nukdokplex-site) after upgrading chirpy to 6.4.2.

### Steps To Reproduce

1. Update Gemfile like this:
```gemfile
# frozen_string_literal: true

source "https://rubygems.org"

gem "jekyll-theme-chirpy", "~> 6.4", ">= 6.4.2"

group :test do
  gem "html-proofer", "~> 4.4"
end

# Windows and JRuby does not include zoneinfo files, so bundle the tzinfo-data gem
# and associated library.
platforms :mingw, :x64_mingw, :mswin, :jruby do
  gem "tzinfo", ">= 1", "< 3"
  gem "tzinfo-data"
end

# Performance-booster for watching directories on Windows
gem "wdm", "~> 0.1.1", :platforms => [:mingw, :x64_mingw, :mswin]

# Lock `http_parser.rb` gem to `v0.6.x` on JRuby builds since newer versions of the gem
# do not have a Java counterpart.
gem "http_parser.rb", "~> 0.6.0", :platforms => [:jruby]
```
This new Gemfile was copied from chirpy-starter's one.

2. Commit

### Expected Behavior

Normal dependencies installation

### Environment

```yml
name: "Build and Deploy"
on:
  push:
    branches:
      - main
      - master
    paths-ignore:
      - .gitignore
      - README.md
      - LICENSE

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

permissions:
  contents: read
  pages: write
  id-token: write

# Allow one concurrent deployment
concurrency:
  group: "pages"
  cancel-in-progress: true

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v4
        with:
          fetch-depth: 0
          # submodules: true
          # If using the 'assets' git submodule from Chirpy Starter, uncomment above
          # (See: https://github.com/cotes2020/chirpy-starter/tree/main/assets)

      - name: Setup Pages
        id: pages
        uses: actions/configure-pages@v3

      - name: Setup Ruby
        uses: ruby/setup-ruby@v1
        with:
          ruby-version: 3
          bundler-cache: true

      - name: Build site
        run: bundle exec jekyll b -d "_site${{ steps.pages.outputs.base_path }}"
        env:
          JEKYLL_ENV: "production"

      - name: Test site
        run: |
          bundle exec htmlproofer _site \
            \-\-disable-external=true \
            \-\-ignore-urls "/^http:\/\/127.0.0.1/,/^http:\/\/0.0.0.0/,/^http:\/\/localhost/"

      - name: Upload site artifact
        uses: actions/upload-pages-artifact@v2.0.0
        with:
          name: site
          path: "_site${{ steps.pages.outputs.base_path }}"
          retention-days: 1

  deploy:
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: ubuntu-latest
    needs: build
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v2
        with:
          artifact_name: site
          timeout: 180000
```

### Anything else?

[https://github.com/nukdokplex/nukdokplex-site/actions/runs/7593435149/job/20683909405](https://github.com/nukdokplex/nukdokplex-site/actions/runs/7593435149/job/20683909405)
```
/opt/hostedtoolcache/Ruby/3.3.0/x64/bin/bundle config --local path /home/runner/work/nukdokplex-site/nukdokplex-site/vendor/bundle
/opt/hostedtoolcache/Ruby/3.3.0/x64/bin/bundle lock
Fetching gem metadata from https://rubygems.org/...........
Resolving dependencies...
Writing lockfile to /home/runner/work/nukdokplex-site/nukdokplex-site/Gemfile.lock
Cache key: setup-ruby-bundler-cache-v6-ubuntu-22.04-x64-ruby-3.3.0-wd-/home/runner/work/nukdokplex-site/nukdokplex-site-with--without--only--Gemfile.lock-afa1f10d6b239b8f26aa65ee903c6459ede683ff901e8893ba2480e29f93beff
/opt/hostedtoolcache/Ruby/3.3.0/x64/bin/bundle install --jobs 4
Fetching gem metadata from https://rubygems.org/.........
Fetching public_suffix 5.0.4
Fetching colorator 1.1.0
Fetching concurrent-ruby 1.2.3
Fetching eventmachine 1.2.7
Installing colorator 1.1.0
Installing public_suffix 5.0.4
Fetching http_parser.rb 0.8.0
Installing eventmachine 1.2.7 with native extensions
Installing concurrent-ruby 1.2.3
Fetching ffi 1.16.3
Installing http_parser.rb 0.8.0 with native extensions
Installing ffi 1.16.3 with native extensions
Fetching forwardable-extended 2.6.0
Installing forwardable-extended 2.6.0
Fetching google-protobuf 3.25.2
Installing google-protobuf 3.25.2 with native extensions
Fetching mercenary 0.4.0
Installing mercenary 0.4.0
Fetching racc 1.7.3
Installing racc 1.7.3 with native extensions
Fetching parallel 1.24.0
Installing parallel 1.24.0
Fetching rainbow 3.1.1
Installing rainbow 3.1.1
Fetching yell 2.2.2
Installing yell 2.2.2
Fetching zeitwerk 2.6.12
Installing zeitwerk 2.6.12
Fetching rb-fsevent 0.11.2
Installing rb-fsevent 0.11.2
Fetching rexml 3.2.6
Installing rexml 3.2.6
Fetching liquid 4.0.4
Installing liquid 4.0.4
Fetching rouge 4.2.0
Installing rouge 4.2.0
Fetching safe_yaml 1.0.5
Installing safe_yaml 1.0.5
Fetching unicode-display_width 2.5.0
Installing unicode-display_width 2.5.0
Fetching webrick 1.8.1
Installing webrick 1.8.1
Fetching jekyll-paginate 1.1.0
Installing jekyll-paginate 1.1.0
Fetching addressable 2.8.6
Installing addressable 2.8.6
Fetching i18n 1.14.1
Installing i18n 1.14.1
Fetching pathutil 0.16.2
Installing pathutil 0.16.2
Fetching nokogiri 1.16.0 (x86_64-linux)
Installing nokogiri 1.16.0 (x86_64-linux)
Fetching kramdown 2.4.0
Installing kramdown 2.4.0
Fetching terminal-table 3.0.2
Installing terminal-table 3.0.2
Fetching kramdown-parser-gfm 1.1.0
Installing kramdown-parser-gfm 1.1.0
Fetching ethon 0.16.0
Fetching rb-inotify 0.10.1
Installing rb-inotify 0.10.1
Installing ethon 0.16.0
Fetching listen 3.8.0
Installing listen 3.8.0
Fetching jekyll-watch 2.2.1
Installing jekyll-watch 2.2.1
Fetching typhoeus 1.4.1
Installing typhoeus 1.4.1
Fetching html-proofer 4.4.3
Installing html-proofer 4.4.3
Fetching em-websocket 0.5.3
Installing em-websocket 0.5.3
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory:
/home/runner/work/nukdokplex-site/nukdokplex-site/vendor/bundle/ruby/3.3.0/gems/google-protobuf-3.25.2/ext/google/protobuf_c
rake
RUBYARCHDIR\=/home/runner/work/nukdokplex-site/nukdokplex-site/vendor/bundle/ruby/3.3.0/extensions/x86_64-linux/3.3.0/google-protobuf-3.25.2
RUBYLIBDIR\=/home/runner/work/nukdokplex-site/nukdokplex-site/vendor/bundle/ruby/3.3.0/extensions/x86_64-linux/3.3.0/google-protobuf-3.25.2
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/rubygems.rb:259:in
`find_spec_for_exe': can't find gem rake (>= 0.a) with executable rake
(Gem::GemNotFoundException)
from /opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/rubygems.rb:278:in
`activate_bin_path'
	from /opt/hostedtoolcache/Ruby/3.3.0/x64/bin/rake:25:in `<main>'

rake failed, exit code 1

Gem files will remain installed in
/home/runner/work/nukdokplex-site/nukdokplex-site/vendor/bundle/ruby/3.3.0/gems/google-protobuf-3.25.2
for inspection.
Results logged to
/home/runner/work/nukdokplex-site/nukdokplex-site/vendor/bundle/ruby/3.3.0/extensions/x86_64-linux/3.3.0/google-protobuf-3.25.2/gem_make.out

/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/rubygems/ext/builder.rb:125:in
`run'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/rubygems/ext/rake_builder.rb:30:in
`build'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/rubygems/ext/builder.rb:193:in
`build_extension'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/rubygems/ext/builder.rb:227:in
`block in build_extensions'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/rubygems/ext/builder.rb:224:in
`each'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/rubygems/ext/builder.rb:224:in
`build_extensions'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/rubygems/installer.rb:852:in
`build_extensions'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundler/rubygems_gem_installer.rb:76:in
`build_extensions'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundler/rubygems_gem_installer.rb:28:in
`install'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundler/source/rubygems.rb:205:in
`install'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundler/installer/gem_installer.rb:54:in
`install'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundler/installer/gem_installer.rb:16:in
`install_from_spec'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundler/installer/parallel_installer.rb:132:in
`do_install'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundler/installer/parallel_installer.rb:123:in
`block in worker_pool'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundler/worker.rb:62:in
`apply_func'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundler/worker.rb:57:in
`block in process_queue'
  <internal:kernel>:187:in `loop'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundler/worker.rb:54:in
`process_queue'
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundler/worker.rb:90:in
`block (2 levels) in create_threads'

An error occurred while installing google-protobuf (3.25.2), and Bundler cannot
continue.

In Gemfile:
  jekyll-theme-chirpy was resolved to 6.4.2, which depends on
    jekyll-archives was resolved to 2.2.1, which depends on
      jekyll was resolved to 4.3.3, which depends on
        jekyll-sass-converter was resolved to 3.0.0, which depends on
          sass-embedded was resolved to 1.70.0, which depends on
            google-protobuf
Error: The process '/opt/hostedtoolcache/Ruby/3.3.0/x64/bin/bundle' failed with exit code 5```

**Comment by nukdokplex** [comment]
fixed by adding minor version **"3.2"** to workflow file according to this commit: [1efe631](https://github.com/cotes2020/chirpy-starter/commit/1efe631e7ed0b977f773f243233312335103162f)

**Comment by brootware** [comment]
Thanks for this @nukdokplex ! Was running into the same error and this fixed it!
